packages/git/src/browser/index.ts

```diff
@@ -0,0 +1,40 @@
+/********************************************************************************
+ * Copyright (C) 2018 Arm and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { FileStat } from '@theia/filesystem/lib/common';
+import { Git, Repository } from '../common';
+
+export const ScmWidgetFactory = Symbol('ScmWidgetFactory');
+
+export const GIT_WIDGET_FACTORY_ID = 'git';
+
+export interface ScmWidgetFactory {
```

Comment:
The name suggest it creates ScmWidget instances. But I don't see a method that would do that.

Reply:
> > Also what is the use case of having multiple scm containers in one view?
> 
> IIUC, in this case it would reuse the same view, but the actual widget inside it would be VCS dependent. And I also seem to understand that one assumption made in this change is that their will be only one VCS type/kind per folder? As in, a folder versioned using git won't use mercurial at the same time?

Yes, correct.  Only one ScmWidget visible at a time, they are stacked.  It would be possible, I suppose, for two SCM implementions to claim they support a repository in the same directory.  If this happens the first one will be used.